src/collectors/ip/ip.py

```diff
@@ -56,30 +69,38 @@ def get_default_config(self):
         config = super(IPCollector, self).get_default_config()
         config.update({
             'path': 'ip',
-            'allowed_names': 'InAddrErrors, InDelivers, InDiscards, ' +
-            'InHdrErrors, InReceives, InUnknownProtos, OutDiscards, ' +
-            'OutNoRoutes, OutRequests'
+            'allowed_names': ','.join(self.DEFAULT_METRICS)
         })
         return config
 
+    def open_file(self, filepath):
```

Comment:
does this really need to be abstracted out into a separate function? Can'e you directly issue an open and catch the permission exception?